nengo/conftest.py

```diff
@@ -269,6 +269,57 @@ def seed(request):
     return function_seed(request.function, mod=TestConfig.test_seed)
 
 
+@pytest.fixture
+def allclose(request):
+    """A functional replacement for np.allclose.
+
+    This allows backends to specify different tolerances by adding
+
+    .. code-block:: ini
+
+        nengo_test_tolerances =
+            path/to/test/file:testname atol=x rtol=y
+            path/to/other/file:testname2 atol=z  # this is a comment
+            ...
+
+    to their pytest config file.
+
+    If a test contains multiple ``allclose`` checks then multiple entries
+    with the same testname can be added to the config, and will be applied
+    in the given order.
+    """
+
+    call_count = [0]  # need to access the closure this way for 2.7 compat
+    override = []
+    for line in request.config.getini("nengo_test_tolerances"):
+        split_line = line.split()
+        testname = "*" + split_line[0]
+        nodename = get_item_name(request.node)
+        if fnmatch(nodename, testname):
+            kwargs = {}
+            for entry in split_line[1:]:
+                if entry.startswith("#"):
+                    break
+                k, v = entry.split("=")
+                kwargs[k] = float(v)
+            override.append(kwargs)
+
+    def _allclose(a, b, **kwargs):
+        # set tolerances
+        if override:
+            kwargs.update(override[min(call_count[0], len(override) - 1)])
```

Comment:
So if `call_count` is outside the number of overrides, then we just keep applying the last one? Maybe we should state this in the docstring.

Reply:
Also, just for me to understand, with parametrized tests, there's no way of explicitly targeting a particular parameter setting? The way to do that would be to have multiple entries for the test in the order that the parameters are used?

One thing that's a little confusing about this is when a test is parametrized multiple times. For example
```
pytest.mark.parametrize('a', [0, 1])
pytest.mark.parametrize('b', [2, 3])
def test_my_stuff(a, b):
    print((a, b))
```
I assume this will print "(0, 2) (0, 3) (1, 2) (1, 3)" but I'm not sure.

I'm wondering if there's a better way to do this. I think the `request` does include information about the parameters, so maybe there's a way to check this?

Reply:
Yes you can explicitly target parameter settings with e.g.
```
nengo_test_unsupported = file/path/testname[parameter_value]
```
I added that to the tests/documentation so it's more obvious.  I also added a thing to automatically escape special characters, so that people don't have to remember to do that.